app/src/main/java/com/github/se/polyfit/ui/screen/MapScreen.kt

```diff
@@ -114,39 +136,103 @@ fun MapScreen(
       contentAlignment = Alignment.BottomStart) {
         GoogleMap(
             cameraPositionState = cameraPositionState,
-            properties = MapProperties(isMyLocationEnabled = true)) {
-              Circle(
-                  center = LatLng(currentLocation.value.latitude, currentLocation.value.longitude),
-                  radius =
-                      (sliderPosition).toDouble() + 10, // Make the shadow circle slightly larger
-                  strokeColor =
-                      Color.Black.copy(alpha = 0.2f) // Use a semi-transparent color for the shadow
-                  ,
-                  visible = isCircleShown)
-              Circle(
-                  center = LatLng(currentLocation.value.latitude, currentLocation.value.longitude),
-                  radius = (sliderPosition).toDouble(),
+            properties = MapProperties(isMyLocationEnabled = true),
+            onMapClick = { selectedPost = null }) {
+              val latLng = LatLng(currentLocation.value.latitude, currentLocation.value.longitude)
+              val doubleSliderPosition = sliderPosition.toDouble()
+              CircleOnMap(
+                  center = latLng,
+                  radius = doubleSliderPosition,
                   strokeColor = PrimaryPurple,
                   visible = isCircleShown)
-              listMarker.forEach { markerState ->
-                Marker(
-                    state = markerState,
-                    title = "title",
-                    contentDescription = "description",
-                    onClick = { goToMarker(goToPost, markerState) })
-              }
+              CircleOnMap(
+                  center = latLng,
+                  radius = doubleSliderPosition + 10,
+                  strokeColor = Color.Black.copy(alpha = 0.2f),
+                  visible = isCircleShown)
+
+              Marker(listMarker) { post -> selectedPost = post }
```

Comment:
I'm not sure if the order affect who is on the foreground but the marker should be above the circle on the screen

Reply:
No, it doesn't affect